src/main/java/io/vertx/core/logging/LoggerFactory.java

```diff
@@ -46,18 +47,54 @@ public static synchronized void initialise() {
     }
 
     if (className != null) {
-      ClassLoader loader = Thread.currentThread().getContextClassLoader();
+      ClassLoader tcccl = Thread.currentThread().getContextClassLoader();
       try {
-        Class<?> clz = loader.loadClass(className);
-        delegateFactory = (LogDelegateFactory) clz.newInstance();
-      } catch (Exception e) {
-        throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
+        newDelegateFactory = loadClass(className, tcccl);
+      } catch (IllegalArgumentException e) {
+        ClassLoader thisClasssLoader = LoggerFactory.class.getClassLoader();
+        newDelegateFactory = loadClass(className, thisClasssLoader);
       }
     } else {
-      delegateFactory = new JULLogDelegateFactory();
+      newDelegateFactory = new JULLogDelegateFactory();
+    }
+
+    LoggerFactory.delegateFactory = newDelegateFactory;
+  }
+
+  private static LogDelegateFactory loadClass(String className, ClassLoader classLoader) {
+    try {
+      Class<?> clz = classLoader.loadClass(className);
+      return (LogDelegateFactory) clz.newInstance();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Error instantiating LogDelegateFactory implemention: \"" + className + "\"", e);
```

Comment:
There is a few issues here:

* The class cannot be loaded
* The access is not allowed 
* The constructor throws an exception

We may want to throw different type of exceptions for these types.